### ISSUE TITLE
Need some conversion functions from Cartesian to Spherical coordinates...

### ISSUE THREAD

**Issue by Felix-neko** [scipy/scipy#7943]
I have an array of cartesian (x, y, z)-points. I want to convert it into a vector of spherical (r, azimuth, elevation)-points.

Can you add something for this in future releases?

**Comment by ilayn** [comment]
Related discussion on https://github.com/numpy/numpy/issues/5228

**Comment by tylerjereddy** [comment]
As the discussion in the linked thread implies, this is indeed tricky business with the various different conventions, and the rather large number of possibly-useful coordinate system transformation implementations that a community might want (how to draw the line / decide what is appropriate for the library). I usually end up having to write / use my own small functions for this.

**Comment by zofy** [comment]
Ok, guys, can I take care of it?

**Comment by person142** [comment]
> Ok, guys, can I take care of it?

It might be good to wait until there is more consensus on the issue. (Don't want hard work to go to waste.)

**Comment by Felix-neko** [comment]
Yes, that will really be useful!
: 3

**Comment by zofy** [comment]
Ok then, so you say it is better to do nothing for now.... But when will there be more consesus?

**Comment by Felix-neko** [comment]
No-no-no, I've said that adding such a conversion will be really useful!

P.S. Pardon my English...

**Comment by SoumyadeepThakur** [comment]
Any updates on this thread?

**Comment by zofy** [comment]
Alright, I almost forgot about it. So, I'll create a PR and we'll see whether it's gonna be added there or not.

**Comment by rgommers** [comment]
The numpy issue linked above does seem very relevant. Polar coordinates are more straightforward, spherical is unclear. Then the question is where they would go - perhaps in `spatial.transform`?

**Comment by Make42** [comment]
I would also be very interested if this could be done for high dimensional coordinate systems, esp. spherical <-> cartesian for systems with dimension larger than 3.

**Comment by ThomasHilger** [comment]
I also think this should be in scipy.
Regarding the convention, I think it should naturally follow the convention used by other modules, like the spherical harmonics:
https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.sph_harm.html
Also: by choosing/setting a convention, work will be become easier and more clear, as everybody then knows what it is and can customize from there on.

**Comment by Kemsekov** [comment]
I have a piece of code in c# that does this transformation for n - dimensional vectors, it is pretty obviously portable to c, and hence, in scipy

https://github.com/Kemsekov/Hyperspherical-Coordinates-transformation
